### PR TITLE
kinematicwalking

### PR DESCRIPTION
**Pull request by aremazeilles**

### CODE REVIEW

**Comment by aremazeilles** [comment]
Hi @juritaborri 

You can look at the changes done with the button file changed. 
Please have a look at the changes proposed.


If you have question, we can discuss. If not, Press "Merge pull request" button. That will bring it to the master branch.

I would then ask you to perform similar operation on the other scripts. But we can chat on it 
If you prefer me to help (like for moving the file and so on in the git repo), just tell me.

File: tests/data/output/Protocol1/pi_mrom.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+type: 'labelled_matrix'
+measure unit: '%'
```
Comment (juritaborri) [inline tests/data/output/Protocol1/pi_mrom.yaml:2]: These are degree '°'

**Comment by juritaborri** [comment]
OK!
I will check all the canges and then come back to you.
I also would try to make similar changes in other scripts to verify if I am able to use in the right way the git.

**Comment by aremazeilles** [comment]
If needed, we can make a short phone call to explain some aspects, if needed

File: beat/Functions/kinematic_routine_walking.m [hunk 2]
```diff
@@ -95,51 +95,52 @@
 [aaa, name, extension]=fileparts(PlatformData);
 name2=regexprep(name,'_PlatformData','');
 file_id=fopen(strcat(outFolder,"/", name2, "_mROM", ".yaml"),'w'); %%open file to write into
-fprintf(file_id, "type: 'vector'\n");
-fprintf(file_id, "'measure unit: %%' \n");
-label_str="angle_label: [";
+fprintf(file_id, "type: 'labelled_matrix'\n");
+fprintf(file_id, "measure unit: '%%'\n");
```
Comment (juritaborri) [inline beat/Functions/kinematic_routine_walking.m]: It is wrong also in the previous version, it must be degree °

File: beat/Functions/kinematic_routine_walking.m [hunk 3]
```diff
@@ -92,9 +92,7 @@
 angle_label2=cat(2,angle_label(angle_r),angle_label(angle_l));
 
 %%save mROM value in .yaml file
-[aaa, name, extension]=fileparts(PlatformData);
-name2=regexprep(name,'_PlatformData','');
-file_id=fopen(strcat(outFolder,"/", name2, "_mROM", ".yaml"),'w'); %%open file to write into
+file_id = fopen(strcat(outFolder, "/pi_mrom.yaml"),'w'); %%open file to write into
```
Comment (juritaborri) [inline beat/Functions/kinematic_routine_walking.m:91]: I am not sure that it the best solution for saving the pi. I mean, the mROM can be obtained with different protocol. Maybe it should be useful to put in the file name also the protocol number.
Another solution could be to put the name in the outFolder, but I believe that it is less automatic.

File: beat/Functions/kinematic_routine_walking.m [hunk 4]
```diff
@@ -121,7 +119,7 @@
 
 
 %%save CoV value in .yaml file
-file_id=fopen(strcat(outFolder,"/", name2, "_CoV", ".yaml"),'w'); %%open file to write into
+file_id=fopen(strcat(outFolder,"/pi_cov.yaml"),'w'); %%open file to write into
```
Comment (juritaborri) [inline beat/Functions/kinematic_routine_walking.m:116]: Similar to the mROM

File: beat/Functions/kinematic_routine_walking.m (hunk 3 above)
Comment (aremazeilles) [reply beat/Functions/kinematic_routine_walking.m:91]: The initial code made some assumption on the way the input filename wasstructured. We should not.
Your code generate a pi_mrom, we store it as is. It is up to the caller of your routines to make this relation. 

And actually, this is why we suggest to store all output in folder, previously created. This way, the caller can create folder protocol_i, and ask to store results there.

**Review by juritaborri — COMMENTED** [review]
Perfect!

**Comment by juritaborri** [comment]
I have adjusted all the other functions based on your suggestions provided for kinematicwalking.
I have also updated the format of some outputs and the readme file.

**Comment by aremazeilles** [comment]
Ok, thanks. I´ll have a look at it along the week.

**Comment by aremazeilles** [comment]
Are you trying the files under octave on your machine? 
Perhaps this is because you use it from the gUI, so that what is loaded remain loaded somehow, but none of them worked to me. I had to add the pkg load in each of the run_ files

**Comment by aremazeilles** [comment]
I will merge this, and open some issues for other aspects.

You should swith to the master branch, and make fetch / merge
Then, if you want to solve an issue, create a branch (one issue per branch), and commit there. From time to time, push it to a new remote branch.

Is that ok with you?

File: beat/Functions/kinematic_routine_walking.m (hunk 2 above)
Comment (aremazeilles) [reply beat/Functions/kinematic_routine_walking.m]: SI would be better, i.e. radians. But let's leave it as is for later on.

Comment (juritaborri) [reply beat/Functions/kinematic_routine_walking.m]: When you are looking at human angles, it is usually expressed in ° (99.9% of literature). However, if you want we can express it in radians

**Comment by juritaborri** [comment]
> Are you trying the files under octave on your machine?
> Perhaps this is because you use it from the gUI, so that what is loaded remain loaded somehow, but none of them worked to me. I had to add the pkg load in each of the run_ files

Yes.. I run them always with OCtave GUI. How can I solve this issue?

**Comment by aremazeilles** [comment]
I solved the issue, by adding in each run file the load of the io and signal packages

**Pull request merged**